src/views/transaction/components/transaction-form/transaction-form.view.jsx

```diff
@@ -165,8 +165,11 @@ function TransactionForm({
   }
 
   function handleAmountChange(data) {
-    setAmount(data.amount.tokens);
     setShowInFiat(data.showInFiat);
+
+    if (data.isUntouched) return;
```

Comment:
Instead of creating more branches in the code (if...else) which makes it more difficult to later find bugs, I will create a `const [isAmountInputDirty, setIsAmountInputDirty] = React.useState()` and then here just set the value: `setIsAmountInputDirty(data.isDirty)`. Later when you decide to display the error or not you just render it in case `isAmountInputDirty === true && isAmountInvalid === true`.